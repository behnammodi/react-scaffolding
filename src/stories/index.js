import { storiesOf } from '@storybook/react';
import { action as addOnAction } from '@storybook/addon-actions';
import {
  withKnobs,
  text as typeText,
  boolean as typeBoolean,
  number as typeNumber
} from '@storybook/addon-knobs';

export const add = (name, stories) => {
  const of = storiesOf(name, module);
  of.addDecorator(withKnobs);
  stories.forEach(story =>
    of.add(story.name, story.render, { notes: story.notes })
  );
};

export const action = addOnAction;
export const text = typeText;
export const boolean = typeBoolean;
export const number = typeNumber;
