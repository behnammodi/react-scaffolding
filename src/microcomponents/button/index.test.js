import { snapshot } from 'test';
import stories from './index.stories';

snapshot(stories);
