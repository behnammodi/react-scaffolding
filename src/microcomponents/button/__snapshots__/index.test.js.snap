// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with action 1`] = `
<button
  className="button-0-1-1"
  disabled={false}
  onClick={[Function]}
>
  Hello IB
</button>
`;

exports[`Renders correctly with disabled 1`] = `
<button
  className="button-0-1-1"
  disabled={true}
>
  Hello IB
</button>
`;

exports[`Renders correctly with text 1`] = `
<button
  className="button-0-1-1"
>
  Hello IB
</button>
`;
